### ISSUE TITLE
New Site Import - Select Call Centre

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#305]
In GitLab by @kdjstudios on Apr 14, 2017, 14:11

This ticket will detail the exported files for this new Site/Client and how to import their existing data into SAB.

#### [Effort Log for #305 ](http://gitlab.aavaz.biz/AnswerNet/SABilling/wikis/effort-log-305)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 24, 2017, 11:16

Update:

Hi Kyle,

Please use this password to open files.

SCC2017

Let me know if you have any issues. Thanks. Garrett. 

[Data_from_SCC_.dmg](/uploads/c90d8308c3ca8f3a2a0b94631795a1cd/Data_from_SCC_.dmg)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 24, 2017, 14:11

Here is the data in a zip format. Garrett Also included a brief explanation of the exports.

[Data_4_SA_Billing_2.zip](/uploads/62e2de2f641ad3f32e5ecbbf89dd3f77/Data_4_SA_Billing_2.zip)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 24, 2017, 14:13

@tim.traylor @abakshi @smasih @rchauhan 

Is everyone available for a call tomorrow (4/25/17) to discuss the export data; and generate any questions or additional requests for Garrett?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 24, 2017, 14:51

@smasih @abakshi @pchaudhary 

Update: Tim would like for your team to review the attached files and generate a list of questions/concerns before we have a call. Please complete this and we will schedule a call accordingly.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Apr 25, 2017, 06:25

@kdjstudios Hi Kyle, Did we provide the content and relationship information we got during the client meeting?  Also, we have to explain the breakout of that field with the three different embedded values.  Sorry, I was rushing yesterday or would have thought to make sure we had already given this to the guys.  Thx, Tim

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 25, 2017, 08:16

@tim.traylor @smasih @abakshi @rchauhan @pchaudhary 

Garrett provided a very detailed explanation document of his billing data along with the data export. We are awaiting for the aavaz team to read it and then ask questions.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 27, 2017, 10:06

Tim's Notes from Join.Me on parsing the Tas1 file.

Right to left: last 2 are patch, next 3 are per minute overage, remaining 2 or 3 are allowance.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 1, 2017, 15:50

UPDATE:

His Billing is on the May 3rd. Since we are still working on putting together a method/script to import his data I informed him we expect his time line would be the following:

By May 24th: We will have imported his information and provide him with full manager access to his data.

By May 31st: We will have the import scripts tested and verified working. I will also have gone through training of the system. If possible we will try to run a side by side of his billing to find any major errors.

By June 28th: We will run a side by side billing in his current system and in SAB to find any discrepancies. 

On July 26th: We will have fixed all issues and he will be using the SA Billing Platform from now on.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on May 2, 2017, 07:07

@kdjstudios @abakshi Hi Kyle, we should ask him for a full export before he posts 5/3, request the usage file(s) that he does post, and then get the file with the balances following his 5/3 processing.  

This will allow the guys to load the pre-5/3 data, test the posting of the usage, and compare to his system's resulting balances. We should realistically be able to parallel for 6/3 and be live for 7/3 (assuming all went well).

Thanks,
Tim

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 2, 2017, 12:44

@tim.traylor @abakshi @smasih 

I sent out a request to Garrett this morning. He stated that there should not have been any changes since the last full export he sent over. He also did provide us with the April Billing file.

Explanation of file: 

The file has some auto generated information in the first 40ish rows that are unused and to be ignored. These are easily spotted because they do not have a correct "Handle ID".

> Handle ID is the first column. There should be 5 rows per each Handle ID.

> Type is the second column. This tells us if it is the call time, patch time, counts, etc.

> Value is the third column. This is the actually count or metric value for each type.

> Start Date is the fourth column. This is the start date of the usage period. 

[TUSACC04.csv](/uploads/cd993d570deab75c4622279f3e7ea49c/TUSACC04.csv)

I will continue to see if he is able to export the resulting balances after he runs his billing.

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on May 3, 2017, 02:14

@smasih @pchaudhary 

Let's check the understanding, impact  and questions here.

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on May 3, 2017, 08:11

@smasih 

Two important deliverables required here which can be added on WIKI link:
* RAID log (Risk, Assumption, Issues, Dependencies)
* Capture the efforts on daily basis in comments on this ticket itself, using the following format:

| Task ID| Task | Time Spent (h) | Time Remaining (h) |
| ---: | :--: | -------------: | -----------------: |
| 1    | abc  | 1              | 1                  |

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 3, 2017, 08:28

@abakshi @smasih @pchaudhary 

I just looked at both dev and Hosted. IT appears we have not imported any data yet; customers and accounts. May we have an update on the import?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on May 3, 2017, 08:42

@kdjstudios @tim.traylor @rchauhan @pchaudhary 

@abakshi  I have started working on it, it's a lot different than the previous Import we did for NEIT therefore there are no question at the moment but yes  
over the course of development I will be having many.

**As per my best understanding these are the tasks/todo's we have to do in order to complete the Import:**
* Create Site
  * Add Ldap security groups and legal Entity   
* Import Accounts and Customers (`2-3 days`)
  * Customers
  * Accounts
* Add Activities on Site (`1hr`)
* Import activities on account level  (`1 day`)
* Test and Check the Import (`0.5day`)
* Create Posting Script matching to it's Usage File (`1-2 days`) 
* Run test billing (`0.5 day`)
  * Check/verify calculations
* Run the Final Billing. (`1hr`)
* Deploy (`1hr`)

**So the  effort estimate is approx  _6_ days**

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on May 4, 2017, 09:42

All external customers are not using LDAP but they use system-level security.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on May 8, 2017, 04:54

@kdjstudios  @abakshi  @tim.traylor 

#### [Effort Log for #305 ](http://gitlab.aavaz.biz/AnswerNet/SABilling/wikis/effort-log-305)

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on May 8, 2017, 08:02

@smasih Thanks for the effort log. Now we just need to keep it updated on regular basis.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 8, 2017, 08:18

Here is the latest email from Garrett Bidewell:

So here is the latest export for the cmr1 file, this has up to date canceled accounts and current accounts. as well as balances etc. As well as the latest usage file from szeto. 

one change as taken place, under term reason, we've added a type call hold n/c. representing a account that is "on hold" with no charges attached. 

let me know if you have questions. 

[cmr1-05_03_2017.csv](/uploads/f22993e3352e0222553138de64adb73d/cmr1-05_03_2017.csv)

[TUSACC05.csv](/uploads/2c635f16f98509aea13685943439b18a/TUSACC05.csv)

**Comment by kdjstudios** [comment]
In GitLab by @smasih on May 8, 2017, 09:27

@tim.traylor 

I was able to import all Customers (*only customers not Accounts*) from the old CSV file, but now we have a new one so modifying this one with the modification we did on last import csv file, we will be  good to go.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on May 10, 2017, 09:00

@kdjstudios @abakshi 

@tim.traylor, please review the update on [effort log](http://gitlab.aavaz.biz/AnswerNet/SABilling/wikis/effort-log-305).

Now, moving forward we need to add activities to site and account would you please point which file it is ..? as per my understanding it is **tas1-04:20:2017**  from the old zipped file Garrett has sent ..??

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 10, 2017, 09:39

@smasih @tim.traylor 

Yes the tas1 file does contain the activities. I believe Garrett had mentioned there may be duplicates in this file.

The tas2 file should contain the associations of the actives in tas1 to individual accounts.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on May 12, 2017, 08:19

@tim.traylor 

Imported additional fees on all accounts using **rcr1.csv**.

#### Questions

* What is the significance of **handle** in csv 
* Billing Period **MXE** was not given in **explanation of billing data 04-20-2017.docx** but was present in **rcr1.csv** so I created one.

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on May 12, 2017, 08:46

@kdjstudios @tim.traylor 

This is currently work in progress. Moved it to the next release milestone.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on May 18, 2017, 06:23

@kdjstudios @tim.traylor 

Need new set of files matching new customer file **cmr1-05_03_2017.csv** , as there are many values that are not matching **tas1** and **tas2** due to which I am not able to import activities on the account level.

I would be great if you can expedite the process because this will help us to end most of the work on Friday.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 18, 2017, 15:30

@smasih @tim.traylor 

I have requested for updated export from Garrett and will follow up again tomorrow if I do not receive a response.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on May 19, 2017, 04:37

@kdjstudios 

thanks kyle

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 22, 2017, 09:06

@smasih 

Still no reply. I followed up again today.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 23, 2017, 11:45

@smasih 

Here are the latest exports from Garrett. 

[MAY03_17_b2tas2.csv](/uploads/8b592d82af2a556c45f5761643353a55/MAY03_17_b2tas2.csv)

[MAY03_17_b2tas1.csv](/uploads/903ba052406fd446766d4d28e4170a44/MAY03_17_b2tas1.csv)

**Comment by kdjstudios** [comment]
In GitLab by @smasih on May 24, 2017, 08:21

@kdjstudios @tim.traylor 

Great, thanks for the files but, we need a newer version of **rcr1.csv** file as well.

Additionally, it would be great if we can get a **usage.csv**  file along with the switch mapping codes and calculations.

thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 24, 2017, 08:34

@smasih 

I will request for a new version of that file also. Did the current one not work on your import attempt?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on May 24, 2017, 08:50

@kdjstudios 

**Yes it did** but, there is difference in accounts between newer and older set.   
Import code for **rcr1.csv** is ready all we need is a new file in order to import updated data matching with the new accounts as per **cmr1-05_03_2017.csv**

thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 24, 2017, 12:41

@smasih 

Here is the file Garrett Just sent me.

[MAY03_17_b2rcr1.csv](/uploads/25acf12318dfe7fc5781e97abea566d7/MAY03_17_b2rcr1.csv)

**Comment by kdjstudios** [comment]
In GitLab by @smasih on May 25, 2017, 05:51

@kdjstudios @tim.traylor 

I just completed import on **dev** for SCC
 * Activities
 * Additional fees
 * Accounts
 * Customers
 * Site

Still to achieve:  
* Posting Script [**dependancy: not received posting usage csv** one that Kyle posted on his last comment is list of customers and accounts. We need usage csv and mapping document.]
* Run Billing Cycle

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on May 30, 2017, 07:19

@tim.traylor @kdjstudios @smasih 

We  can't proceed any further on this unless we can get into a discussion with Tim, sooner (by tomorrow).

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 30, 2017, 10:19

@tim.traylor @abakshi @smasih 

Should I be reaching out to Garrett for an update usage file? I believe he is running his billing tomorrow.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on May 31, 2017, 06:21

@kdjstudios Hi Kyle, I don't believe that last file from 5/3 was actually a usage file - it looks like one of the customer files or something.  You are correct in that we should get the new usage file that he is posting for his billing so we can post and compare.  Thx, Tim

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 31, 2017, 09:12

@tim.traylor @smasih 

I have requested the latest usage file. I will upload here once I have received it.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 1, 2017, 14:33

Here are the two files sent by Garrett. You may need to rename the .fla file to .csv

[TUSACC06.fla](/uploads/4032573a0f75acd7239ce328f005f991/TUSACC06.fla)

[cTwPgl.csv](/uploads/93abae7416f2dee118d2bf8fc8d1f51e/cTwPgl.csv)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 1, 2017, 14:33

@smasih 

Moving back to development.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 2, 2017, 05:56

@kdjstudios

I don't think so if either of them is a usage file, it would very helpful if @tim.traylor you can have a look.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 2, 2017, 09:07

@smasih 

I just changed the .fla file into a .csv file as I stated to do. It is clearly the same format of the previous usage file mentioned in the above comment from me about 25 days ago.  I have renamed the file for you and attached again. 

[TUSACC06.csv](/uploads/303e81fd0a62b2eb029fc3a8e603094b/TUSACC06.csv)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 2, 2017, 10:43

@smasih 

I also tried to create a billing cycle for the site on Dev and was presented with the 'something went wrong error'

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jun 2, 2017, 11:08

@smasih Hi Sumeet, the .fla file, which does need to be renamed to .csv, is the usage file.  The data is vertical and not horizontal like you're used to.  In other words, instead of a single row with with a column for each data element to be counted we have multiple rows for the same account. So there are groups of rows that are different data elements for a single customer.

As an example, look at lines 81 through 85 in the file.  They are entries for account T-4033143964. Each row is a different type of data and the codes in column B are the different types of activities defined for the account.

Thanks,
Tim

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on Jun 6, 2017, 06:43

Here the usage file is different than regular one, i.e. having rows having key and value pair for data instead of columns having values, so its taking more time to process it. 

As discussed during the call today, we may consider selecting the right approach, though there can be more than one ways to process the file:
* Excel formula approach - Sumeet's suggestion
* Hash element approach - Tim's suggestion
* Sort the original file approach - Tim's suggestion
* Follow the same process for parsing and validating the file - Resource ID field based validation - Tim's suggestion

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on Jun 8, 2017, 03:36

The main idea is to complete the monthly processing first by doing data scrubbing manually .  Once the system start monthly processing correctly, we can create a separate parser to automate the manual data scrubbing.

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on Jun 8, 2017, 09:38

Moving this to next release

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 9, 2017, 11:56

@abakshi @tim.traylor 

Do we have and ETA on when Garrett will be able to run this months billing in order for us to do our comparisons?

This was supposed to be ready around the 1st of June I believe; if we are having to wait till the 24th. Then I believe we will have to push back his conversion another month. 

Tim, what is your input on adjusting the timeline?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 12, 2017, 09:51

@tim.traylor @abakshi @rchauhan @smasih 

Do we have any updates on this? As this should have been ready almost two weeks ago to run through a test billing cycle.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 15, 2017, 08:10

@kdjstudios  @tim.traylor 

Updated the [dev](https://dev.sabilling.com/sites/58f76b6997a73759d3000002) with new site **Select Call Centre**
* Imported accounts
* Customers
* Additional fees
* Recurring fees
* Site Activities
* Account activities
* Posting Script
* In order to run monthly billing process 
  * Use usage file we used [scc_usage]([usage_scc.csv](/uploads/2b978f6a742d73b016fe25b92eb87e7d/usage_scc.csv))
  * **There are more than 1400 accounts so monthly process might take a while.**
  * Please Note for the time being we have to manually add a row on top of usage file, look following example (**we have already added this on attached csv file**)
  * ![Screenshot](/uploads/54ecdc315e7a4b2a047d05ffb17da376/Screenshot.png)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 15, 2017, 08:17

@smasih 

When clicking the above link to download the usage file, I am presented with a 404 not found error.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 15, 2017, 09:23

@smasih 

After accepting the missing accounts, and when trying to apply the usage I am presented with the error "The page you were looking for does not exist"

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 15, 2017, 09:41

@kdjstudios 

Ya, I got the error, some how there is a billing period on an account which is not on site level.

**So, I'll get you a usage file with some 10 odd accounts so that you can test the uploading and billing process along with the calculations on invoice  from your end. Mean Time you can just randomly check the Site/Accounts/Customers  on _SAB_ VS the import _CSV_**

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 16, 2017, 08:43

@kdjstudios 

* Still working on exception report, inorder to stop it from repeating itself.
* Attached a [usage file]([test_scc_kyle.csv](/uploads/50ea71adc4cbf0087614a4838bd8b446/test_scc_kyle.csv))
* You can revert, or create new billing cycle one
* Usage file only has 5 accounts out of which 4 are valid and one is for exception report.
* List of account's resource id used in usage CSV

| **resource_id** | **account name**          |
|----------|---------------------------|
| TAG-6128 | 10K Rentals Ltd.          |
| TAC-0967 | 1204 Kensington Road Ltd. |
| TAC-0554 | 1577419 Alberta Ltd.      |
| TAR-3832 | 327241 Alberta Ltd. DBA   |

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 16, 2017, 08:51

@smasih 

Your attached file goes to 404 missing again.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 19, 2017, 07:44

@kdjstudios 

I have e-mailed the file to you.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 19, 2017, 08:27

@smasih 

Thank you, I have received the file and will start with the testing check list today.

What all has already been tested and verified as working?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 19, 2017, 14:47

@smasih @tim.traylor 

Here are my results of what I was able to test so far.

[SCC_6-19.xlsx](/uploads/f79ea07915cb87acd783cd3323b7f06f/SCC_6-19.xlsx)

In regards to the CC and E-Check payments, what information should I be reaching out to Garrett for?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 20, 2017, 01:14

@kdjstudios 

for CC processing there is a new ticket that Tim created #393 so at the moment we don't need any info from Garrett.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 20, 2017, 05:02

@tim.traylor @kdjstudios 

I am still working on exception report, however it won't affect the actual calcuation/usage/invoice.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 23, 2017, 08:49

@kdjstudios  @tim.traylor 

We optimized usage CSV for 8171 from **10m 8sec** to **3m 32s**

Exception report is also working now.

Here is the usage [file]([scc_orignal.csv](/uploads/a82d9bd121761b056e72c51e4d24b3d2/scc_orignal.csv)), feel free to test

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 23, 2017, 08:54

@smasih @tim.traylor 

Your usage file seems to have failed to upload again. This is the third time your upload points to a 404 not found. Maybe try putting it into a zip?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 23, 2017, 09:06

@kdjstudios 

Sorry, maybe it's our git lab server, holdon I'll just email that over to you.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 23, 2017, 09:47

@smasih @tim.traylor 

Here are my overall results of going through a complete billing cycle. Overall it took roughly 15 minutes start to finish. This is a great improvement and I feel this is on par with our other sites/clients. Tim is this an acceptable time? 

* Upload: 3:25 - Shouldn't the screen show a 'processing loading circle?' like the other steps?
* - 314 missing accounts
* Accept: 0:05
* Apply: 5:35
* Finalize: 3:40
* Revert: 4:10

In my opinion I feel we should be ready to run a side by side billing with Garrett, in order to now check for errors and to confirm the calculations/invoices are matching. I believe there next billing cycle will be the 27th. Should I be reaching out for updated exports so all accounts, payments, balances, start off matching?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 26, 2017, 09:38

@smasih @tim.traylor 

Should I be asking Garrett for any updated information?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 27, 2017, 03:19

@kdjstudios 

An updated usage file so that we can start testing the usage/invoice calculation and import.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 27, 2017, 08:44

@smasih @tim.traylor 

Will we also need an updated import files? As I am sure in the last 5 weeks they have accepted payments and the balances have been adjusted.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 29, 2017, 09:18

Here is the updated account balances. 

[cmr1-up2june26-2017.csv](/uploads/6799bd41beb7dcf7b4c54a4c6b05392c/cmr1-up2june26-2017.csv)

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 30, 2017, 24:38

@kdjstudios 

Alright, what I exactly wanted was, invoice total of all accounts from Garret which you can compare with our invoice totals.
The account balance can be different because we ran the process multiple times and of course, there should have been payments on those accounts till now.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 30, 2017, 09:50

@smasih 

Lets check with @tim.traylor ; but I believe we will need to clear out all of our testing cycles, re import the correct account balances from the 26th, then once provided run through a billing cycle with a actual usage file for the 28th. This way we can check our system to Garrett's current system. 

We do not want any incorrect/testing information as this will I am sure lead to confusion, bad reports, etc.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jul 3, 2017, 24:27

@kdjstudios 

Ya, sure we can do this but, I was talking about invoice totals not account balance.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 3, 2017, 08:36

@smasih @tim.traylor 

I just tried to open a few accounts Billing History and Account Ledgers; but was presenting with the something went wrong error. 

I was able to access the usage history and it appears that the import from the 26th was done. I will reach out to Garrett again today for the latest usage file so we may being on running a side by side billing.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 5, 2017, 09:03

@smasih @abakshi @pchaudhary @tim.traylor 

I just received the usage file from Garrett and went to edit the site to change the last billing cycle date appropriately and received the something went wrong error. 

Here is the attached usage file also: [TUSACC07.csv](/uploads/9b1cce13bec56250ac0146f0a60140df/TUSACC07.csv)

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jul 6, 2017, 24:38

@kdjstudios 

what was the last billing cycle date you were trying to enter ..??

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jul 6, 2017, 08:21

@kdjstudios 

We ran the usage report..  
Upload took: 3:47 sec
missing accounts: 313

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 6, 2017, 14:35

@smasih @pchaudhary @tim.traylor 

Upon logging in with Garrett's user account on Dev. I am presented with the 'Something Went Wrong Error" When trying to selecting a customer or Account from the drop down after hitting go. I then tried to log out and now can no longer even access the dev server via Chrome, firefox, or IE.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jul 7, 2017, 01:17

@kdjstudios 

There was space crisis on dev, we had to delete few old backups, hopefully it should be working fine now.
I just changed the password.

username: **gbidewell**  
password: **sunday123**

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 7, 2017, 08:36

@smasih 

Upon changing gbidewell back to a manager in the permissions the same error occured when trying to access the Edit Site page. Please make sure to do your testing with his login as a manager. He needs to have access to the edit pages. 

I also noticed that I am unable to select only a site from the drop down and hit go. - This I think is effecting all users.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jul 7, 2017, 08:58

Hi @kdjstudios  
Sorry for the inconvenience. The Echeck Gateway was missing from the **SCC**, which was causing the issue. You can check now.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jul 7, 2017, 09:07

@kdjstudios   
> I also noticed that I am unable to select only a site from the drop down and hit go. - This I think is affecting all users.  
  

In SAB, the sidebar is for the **Accounts & Customers**. For the **Site** list, you can click on the **Settings** button placed Top-Right.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 7, 2017, 13:53

@pchaudhary 

Thank you for the update and clarification.

@smasih 

The correct billing date is 6/28/17; could we please either have this adjusted, or reverted and rerun billing accordingly. Thanks in advance.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jul 8, 2017, 07:18

@kdjstudios 

Updated as per the request, beside this please let us know if there is anything else we can do to help.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 11, 2017, 08:17

@smasih 

Thank you for updating the last billing cycle date in the site settings. However that did not update any of the invoice dates.

Should we revert the billing cycle date to the 1st, and then re run the billing cycle correctly start to finish with correct dates?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 18, 2017, 15:13

I have not received any update from Garrett to a few of the past emails I have sent him. I also called today a left a message. Will follow up again tomorrow in hopes of obtaining the updated balances so we may proceed in doing our comparisons.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jul 19, 2017, 24:03

thanks for the update Kyle.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 27, 2017, 09:18

Update:

Hi Kyle, Here is updated info as of the 17th of July. 
Billing is today for us, I plan on sending you the same info tomorrow or possibly the day after. (Thursday or Friday) 

[Latest_billing_file_July_17_2017.csv](/uploads/d37eeec3c64022f0756d1092c1689d17/Latest_billing_file_July_17_2017.csv)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 31, 2017, 13:49

@smasih @tim.traylor 

Here are the latest files sent over by Garrett, Please let me know if we need any more files to do the balance comparison and run the next billing cycle. 

[July_26th_2017_usage_report.txt](/uploads/8c8a09d4cd0db9cd70949a9eb6c22d79/July_26th_2017_usage_report.txt)

[July_26th_billing_2017.csv](/uploads/ca61f4645cc0f78a0a08dfc754186544/July_26th_billing_2017.csv)

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Aug 1, 2017, 03:46

@kdjstudios 

Thanks for the files, I have processed the new updated usage file, but aging and account total in **July_26th_billing_2017.csv** won't help, I think we need invoice to invoice comparison so that we could be sure if the calculations and posting script is accurate. So, can you please request for some invoices for **July 26^th billing** for Garrett.... that would be great.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Aug 1, 2017, 06:21

@kdjstudios  Hi Kyle, we had discussed asking them if they have a report that lists their invoices.  Preferably, the detail of each invoice would be listed.  We really need that to compare to our posting results as if we are off anywhere, we won't necessarily know where the difference is unless we have the invoice detail.  Thx, Tim

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 1, 2017, 08:21

@tim.traylor @smasih 

Will we not need to check the account balances first (Using the Billing file) to find accounts that do not match? That way we are not wasting time looking into accounts that do match? Then once we have a list of accounts that do not match we can reach out for the detailed print out of those account invoices?

I will reach out to Garrett for an export of all of his  last months invoices.

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on Aug 8, 2017, 04:08

Waiting for a response from Garrett. Moving to next milestone.

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on Aug 8, 2017, 04:08

Waiting for a response from Garrett. Moving to next milestone.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 18, 2017, 08:31

@abakshi @tim.traylor @smasih 

Here are the  three files Garret was able to send over. Please let us know if you have any questions.

* [Re_Billing_file_as_of_July_17th..msg](/uploads/6b6635de91b057eccd0d5104429c414e/Re_Billing_file_as_of_July_17th..msg)
* [JUL26_17_TAE.csv](/uploads/bf8cd135db7dfa7968ee967ab0405b86/JUL26_17_TAE.csv)
* [dkcSFR.csv](/uploads/93043ef19a16ed357f387b9026c3a355/dkcSFR.csv)
* [dkc7gH.pdf](/uploads/c159cd3cb76a0ba5e21badb83ed31c5b/dkc7gH.pdf)

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Aug 19, 2017, 05:49

Hi [Kyle] ( @kdjstudios  ),   
We have exactly no idea that what these 3 other files are for except the last (*dkc7gH.pdf*). We have checked the last file with some of the accounts & found the matching additional fees.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 19, 2017, 12:46

@pchaudhary @tim.traylor 

Please read the attachment "Re Billing..." as it explains what each file is.

In summary I believe you will be needing to use the dkc7gh.pdf file as it shows each account's invoice and the charges that were on the invoice. 

The "Jul26.." file should be their expoerted usage file 

The "dkc7sfr.csv" file is the export of each accounts 'package' or charges to use to check to insure all accounts have updated rates/charges

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 22, 2017, 08:45

@pchaudhary @smasih @tim.traylor 

Has there been any progress on checking the invoices and balances?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Aug 29, 2017, 04:06

moving to next milestone

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Aug 29, 2017, 05:30

@kdjstudios @tim.traylor 

We picked page **70** of **dkc7gH.pdf** and compared it with the report, most of them were matching completely or partially. We are not able to calculate the activities still need to investigate on that. However, here is our findings.


| Account | Name                       | Reason for not matching                                                                                 |
|---------|----------------------------|---------------------------------------------------------------------------------------------------------|
| 105615  | Precisions rentals         | To be investigated                                                                                      |
| 106939  | AERO, Rental Services      | only zero's in usage row                                                                                |
| 107485  | Aero Rental services (FSJ) | Account was not present in usage file                                                                   |
| 107241  | C Group Energy Inc         | only zeros in usage row                                                                                 |
| 106088  | Versaline Services         | To be investigated                                                                                      |
| 107286  | Thrutubing Solutions       | To be investigated                                                                                      |
| 107637  | CFR Chemicals              | Account was present in usage file but was not present in Import file there fore it's not present in SAB |
| 3930    | Sauder Electric Ltd.       | To be investigated                                                                                      |

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 29, 2017, 10:55

@smasih 

* For 106393	AERO, Rental Services; Correct it appears there was no usage and on page 22 of dkc7gH.pdf the account shows a zero balance. I believe you may have a typo and were looking at account T-4033143994 - 106939 AERO Rental Services which does have usage.

* For 107485	Aero Rental services (FSJ) It appears to be a flat/base rate account only. After looking at the dkcSFR.csv; so I would presume there should be no ussage.

* For 107241	C Group Energy Inc: It appears to be a flat/base rate account, with an interest line?

* For 107637	CFR Chemicals: It appears to possibly be a new account that was included on the latest account export file dkcSFR.csv.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Aug 31, 2017, 06:15

@kdjstudios  @tim.traylor 

there was a bug in Posting script due to which two activity T-STD and T-PTCH were not getting calculated.
After fixing the script and running the usage process again all the existing accounts are matching as per `dkc7gH.pdf`.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 31, 2017, 08:54

@smasih @tim.traylor 

I just ran the Billing Cycle Detail and we are now over it appears

* $375,865.74 - Our Billing Cycle Detail
* $370,843.38 - dkc7gH.pdf

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 31, 2017, 10:40

I just emailed Garrett to request update exports for his latest billing cycle. This way we can compare the latest accounts, charges, and balances while staying current with his system.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 1, 2017, 12:16

Hello Everyone, 
@smasih @rchauhan @tim.traylor @abakshi @pchaudhary 

I just had a great call with Garrett to let him know our invoice balances are about A 1%  ($5000) discrepancy; which we both are excited about, and to retrieve updated files. Below I will highlight my notes and our suggestions.

* First, he did mention that the import file he has been sending us in the csv format (AUG23_17_TAE.csv) he will no longer have access to once they convert, and they will only have access to the fla file (TUSCACC09.fla). The formatting is slightly different, but has the same content. Except it contains one additional entry for each account.  The "C-NALP" entry will need to be ignored. Sorry to say this but we will need to rework the upload/import posting script to handle this file type and format. 
* 
* Second, he sent us an updated account list (AUG23_17_b2cmr1.csv) for us to use. This will basically wipe out existing account information in replace of this. (i don't know if we want to completely clear accounts; or override existing accounts and remove ones not included) Also he did want us to pay attention to and check the 'interest' and 'tax' charges on accounts. Note the tax column 'K' - ' tax_district' there should be two tax rates, one for Alberta at 5% and another for Ontario at a different rate (he didn't have this rate on hand but said will get back to me accordingly). If marked Exempt there is not Tax (Government accounts). The interest is specified in column 'S' - 'supp_int' and is defined as "A" is Yes it is charged and "N" is No it is not charged.
* 
* Lastly, we need to confirm that this will be for the Aug 23rd billing cycle.

Here are the files: Note he is still exporting out the invoices balances and line items and is going to try for a csv export rather than pdf if possible. I will upload once I have it. 

* [TUSACC09.fla](/uploads/35591f5e22ce0627bf7e278d5dbf0d01/TUSACC09.fla)
* [AUG23_17_TAE.csv](/uploads/ddff3af987bbebff074217c50a172e70/AUG23_17_TAE.csv)
* [AUG23_17_b2tas2.csv](/uploads/9343bfe73d5b55657072feee398eaf39/AUG23_17_b2tas2.csv)
* [AUG23_17_b2cmr1.csv](/uploads/3f1c84bdda83596722c414f5feea4a70/AUG23_17_b2cmr1.csv)

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 2, 2017, 03:48

@kdjstudios 

**need `rcr1.csv` for this new set**

which contains recurring fee info.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 4, 2017, 08:37

there is no list of emails, so Email invoice media won't work. we will be needing a list of emails as well.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 4, 2017, 09:16

@kdjstudios 

Import of the new file is completed on dev. Now we are left with additional fee and recurring fee import. One more day would be required.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 5, 2017, 02:10

@kdjstudios 

Need `tas1.csv` as well, however updated the site with old `tas1.csv` file provided in the initial dump.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 5, 2017, 08:31

@smasih 

Thank you, I have reached out to the Garrett for the additional requests.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 6, 2017, 13:46

@smasih 

Here are the invoice balance file.

[Aug_23_inv_detail.xlsx](/uploads/f377a37be1daa3582b9eaf33dc8dc273/Aug_23_inv_detail.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 8, 2017, 12:54

@smasih 

Please let me know if there are any additional files needed. 

* [AUG23_17_b2tas1.csv](/uploads/32f07e753b45eaf28a7fe41a0cc0e2ef/AUG23_17_b2tas1.csv)
* [emails.xlsx](/uploads/4acb3d7a9b459028ba84611b094f39ab/emails.xlsx)
* [AUG23_17_b2rcr1.csv](/uploads/d8b45c846d6eab291016cd5b58b75bce/AUG23_17_b2rcr1.csv)

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 12, 2017, 10:03

@kdjstudios 

I have imported all the new files and now we can run the billing.

Here is the [file](/uploads/df96ad9b7906eb485aa2345b612868a4/aug_23.csv) for aug 23 if got time you can give it a shot or else we will do it first thing in the morning.

thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 12, 2017, 10:39

@smasih @tim.traylor 

Thanks, I performed the following steps:

* I reset the Billing cycle date to 7/26/17 from the site edit page
* I deleted the current billing cycle
* I refreshed the page to have it reflect then the 'next' billing cycle date of 8/25/17
* I clicked 'create' to create the billing cycle.
* Was presented with "something went wrong error"

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 13, 2017, 24:16

@kdjstudios 

We have resolved the error please try again and I am creating a new ticket #468 as this one holds many comments due to which browser get's stuck for a while.

So from now onwards can we please post comments on #468

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Oct 30, 2017, 04:45

moving to next milestone

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on May 8, 2018, 06:05

removed milestone

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 23, 2018, 10:27

Updated Priority to 3 for SCC

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 14, 2018, 14:29

Closing ticket. This will be irrelevant after a new import.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 14, 2018, 14:49

closed
